From 6b9bd6bfd72e6963786f50bcf00a4ef115b4cb71 Mon Sep 17 00:00:00 2001
From: Russell King <rmk+kernel@arm.linux.org.uk>
Date: Wed, 18 Nov 2015 18:13:40 +0000
Subject: [PATCH 151/162] staging: etnaviv: move ownership transfer from buffer
 to gpu layer

Move the transfer of command buffer ownership from the buffer layer
to the GPU layer - we are transferring ownership to the GPU so it
should be at the GPU layer.

Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>
---
 drivers/staging/etnaviv/etnaviv_buffer.c | 5 -----
 drivers/staging/etnaviv/etnaviv_gpu.c    | 5 +++++
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/staging/etnaviv/etnaviv_buffer.c b/drivers/staging/etnaviv/etnaviv_buffer.c
index b9cc743..35eb749 100644
--- a/drivers/staging/etnaviv/etnaviv_buffer.c
+++ b/drivers/staging/etnaviv/etnaviv_buffer.c
@@ -249,11 +249,6 @@ void etnaviv_buffer_queue(struct etnaviv_gpu *gpu, unsigned int event,
 		link_size = extra_size;
 	}
 
-	/* take ownership of cmdbuffer*/
-	submit->cmdbuf->fence = submit->fence;
-	list_add_tail(&submit->cmdbuf->gpu_active_list, &gpu->active_cmd_list);
-	submit->cmdbuf = NULL;
-
 	/* trigger event */
 	CMD_LOAD_STATE(buffer, VIVS_GL_EVENT, VIVS_GL_EVENT_EVENT_ID(event) |
 		       VIVS_GL_EVENT_FROM_PE);
diff --git a/drivers/staging/etnaviv/etnaviv_gpu.c b/drivers/staging/etnaviv/etnaviv_gpu.c
index 03e5da8..e9d6d40 100644
--- a/drivers/staging/etnaviv/etnaviv_gpu.c
+++ b/drivers/staging/etnaviv/etnaviv_gpu.c
@@ -1127,6 +1127,11 @@ int etnaviv_gpu_submit(struct etnaviv_gpu *gpu,
 
 	etnaviv_buffer_queue(gpu, event, submit);
 
+	/* take ownership of cmdbuffer*/
+	submit->cmdbuf->fence = submit->fence;
+	list_add_tail(&submit->cmdbuf->gpu_active_list, &gpu->active_cmd_list);
+	submit->cmdbuf = NULL;
+
 	for (i = 0; i < submit->nr_bos; i++) {
 		struct etnaviv_gem_object *etnaviv_obj = submit->bos[i].obj;
 
-- 
2.6.2

